pkg/analytics/googleanalytics.go

```diff
@@ -0,0 +1,38 @@
+package analytics
+
+import (
+	"crypto/rand"
+	"fmt"
+
+	ga "github.com/jpillora/go-ogle-analytics"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// UUIDGenerator creates a new UUID each time a new user triggers an event
+func UUIDGenerator() string {
+	b := make([]byte, 16)
+	_, err := rand.Read(b)
+	if err != nil {
+		logf.Log.Error(err, "UUID cannot be generated")
```

Comment:
Don't print the error here, You can return the error, to the parent function

Reply:
done @chandankumar4